ingress/controllers/nginx/controller.go

```diff
@@ -797,9 +860,24 @@ func (lbc *loadBalancerController) getPemsFromIngress(data []interface{}) map[st
 		}
 	}
 
+	lbc.secrMetadataLock.Lock()
```

Comment:
do we really need to maintain this map/locks or can we just relist all ingresses on secret update and break on first name match ?